### PR TITLE
test(sync): #MC-241 added test for calendar synchronisation

### PR DESCRIPTION
**Pull request by Alex-Delaveau**
## Describe your changes
Added some unit test for MC-171 and MC-184 changes.
Added powermock dependencies for tests

## Issue ticket number and link
[MC-241](https://jira.support-ent.fr/browse/MC-241)
[MC-184](https://jira.support-ent.fr/browse/MC-184)
[MC-171](https://jira.support-ent.fr/browse/MC-171)
## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

**Review by sjollois — APPROVED** [review]

**Review by flmartineau — APPROVED** [review]

File: src/test/java/net/atos/entng/calendar/services/CalendarServiceImplTest.java [hunk 1]
```diff
@@ -122,4 +123,44 @@ public void testIsDefaultCalendarIfIsNotDefaultCalendar(TestContext context){
         calendarService.isDefaultCalendar(CALENDAR_ID);
     }
 
+    @Test
+    public void testUpdateCalendar(TestContext context) {
```
Comment (BastienBenard) [inline src/test/java/net/atos/entng/calendar/services/CalendarServiceImplTest.java:127]: A revoir pour tester en asynchrone.

File: src/test/java/net/atos/entng/calendar/services/CalendarServiceImplTest.java [hunk 2]
```diff
@@ -122,4 +123,44 @@ public void testIsDefaultCalendarIfIsNotDefaultCalendar(TestContext context){
...
+
+        Mockito.doAnswer(invocation -> {
+            String collection = invocation.getArgument(0);
+            JsonObject query = invocation.getArgument(1);
+            JsonObject updatedCalendar = invocation.getArgument(2);
+            context.assertEquals(collection, expectedCollection);
+            context.assertEquals(query, expectedQuery);
+            context.assertEquals(updatedCalendar, expectedCalendar);
+            return null;
+        }).when(mongo).update(Mockito.anyString(), Mockito.any(JsonObject.class), Mockito.any(JsonObject.class));
```
Comment (BastienBenard) [inline src/test/java/net/atos/entng/calendar/services/CalendarServiceImplTest.java:162]: il y a 4 paramètre dans le update que tu appel, tu as oublié le handler

File: src/test/java/net/atos/entng/calendar/services/EventServiceMongoImplTest.java [hunk 3]
```diff
@@ -58,4 +65,40 @@ public void testListCalendarEvent_Should_ONLY_retrieve_calendar_by_ID(TestContex
         this.eventService.list(calendarId, user, startDate, endDate, null);
     }
 
+    @Test
+    public void testDeleteDatesAfterComparisonDate(TestContext context) {
+        String date = "2022-05-04";
```
Comment (BastienBenard) [inline src/test/java/net/atos/entng/calendar/services/EventServiceMongoImplTest.java:70]: A revoir pour tester en asynchrone.

File: src/test/java/net/atos/entng/calendar/services/EventServiceMongoImplTest.java [hunk 4]
```diff
@@ -58,4 +65,40 @@ public void testListCalendarEvent_Should_ONLY_retrieve_calendar_by_ID(TestContex
...
+            context.assertEquals(collection, expectedCollection);
+            context.assertEquals(query, expectedQuery);
+            return null;
+        }).when(mongo).delete(Mockito.anyString(), Mockito.any(JsonObject.class), Mockito.any(Handler.class));
+
+        eventService.deleteDatesAfterComparisonDate(CALENDAR_ID, date);
+    }
+
+    @Test
+    public void testRetrieveCalendarId(TestContext context) {
```
Comment (BastienBenard) [inline src/test/java/net/atos/entng/calendar/services/EventServiceMongoImplTest.java:89]: A revoir pour tester en asynchrone.

File: src/test/java/net/atos/entng/calendar/services/CalendarServiceImplTest.java [hunk 5]
```diff
@@ -122,4 +124,49 @@ public void testIsDefaultCalendarIfIsNotDefaultCalendar(TestContext context){
...
     }
 
+    @Test
+    public void testUpdateCalendar(TestContext context) {
+        Async async = context.async();
+
+        JsonObject now = MongoDb.now();
+
+
+        JsonObject defaultCalendar = new JsonObject()
```
Comment (alicedraillard) [inline src/test/java/net/atos/entng/calendar/services/CalendarServiceImplTest.java:134]: a renommer, defaultCalendar a une signification particuliere dans Calendar

**Review by BastienBenard — APPROVED** [review]

**Pull request merged**